docs/samples/Microsoft.ML.Samples/Dynamic/WordEmbeddingTransform.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms.Text;
+namespace Microsoft.ML.Samples.Dynamic
+{
+    class WordEmbeddingTransform
+    {
+        public static void ExtractEmbeddings()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var ml = new MLContext();
+
+            // Get a small dataset as an IEnumerable and convert to IDataView.
+            var data = SamplesUtils.DatasetUtils.GetSentimentData();
+            var trainData = ml.Data.ReadFromEnumerable(data);
+
+            // Preview of the data.
+            //
+            // Sentiment    SentimentText
+            // true         Best game I've ever played.
+            // false        ==RUDE== Dude, 2.
+            // true          Until the next game, this is the best Xbox game!
+
+            // Pipeline which goes through SentimentText and normalize it, tokenize it by words, and removes default stopwords.
+            // After all that cleaning, we apply WordEmbedding with GloVeTwitter25D.
+            // 25D means each word mapped into 25 dimensional space, basically each word represented by 25 float values.
+            var wordsPipeline = ml.Transforms.Text.NormalizeText("NormalizedText", "SentimentText")
+                .Append(ml.Transforms.Text.TokenizeWords("Words", "NormalizedText"))
+                .Append(ml.Transforms.Text.RemoveDefaultStopWords("CleanWords", "Words"))
+                .Append(ml.Transforms.Text.ExtractWordEmbeddings("WordEmbeddings", "CleanWords", WordEmbeddingsExtractingTransformer.PretrainedModelKind.GloVeTwitter25D));
```

Comment:
Recommend concat'ing ngrams (specifically bigram+trichargrams). The general winning combo is almost always ngrams with WordEmbeddings, not just WordEmbeddings by themselves.

Reply:
It's not about best practices for text transformations. It's about how to use WordEmbedding transform. How bigrams and trichargrams will help achieve that?

---
In reply to: [252510852](https://github.com/dotnet/machinelearning/pull/2339#discussion_r252510852) [](ancestors = 252510852)

Reply:
I believe the goal is teach users how to make good models. 

Teaching the user to use the text transform is useful in itself, but we should teach them to make useful models. In a run on ~30 text datasets, the winner was always WordEmbedding+NGrams.

Reply:
Purpose of this sample to show how wordembedding work. 
No questions, we need to educate users how to build most useful models, but we should do it here: [https://dotnet.microsoft.com/apps/machinelearning-ai/ml-dotnet](https://dotnet.microsoft.com/apps/machinelearning-ai/ml-dotnet)
While I'm just trying to show off how word embedding works in general which would be displayed here: [https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.textcatalog.extractwordembeddings?view=ml-dotnet](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.textcatalog.extractwordembeddings?view=ml-dotnet)
More stuff I put, more confusing it would be for user.
In order to explain how Word embedding work, you don't need NGrams, they would just bring confusion.
Also right now in cookbook we don't use text transform we use combination of transforms like in my example :
[https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetCookBook.md#how-do-i-train-my-model-on-textual-data](https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetCookBook.md#how-do-i-train-my-model-on-textual-data)


---
In reply to: [252520461](https://github.com/dotnet/machinelearning/pull/2339#discussion_r252520461) [](ancestors = 252520461)